zokrates_core/src/flatten/mod.rs

```diff
@@ -620,59 +664,27 @@ impl<'ast, T: Field> Flattener<'ast, T> {
         e: FlatExpression<T>,
         c: T,
     ) -> FlatExpression<T> {
-        let bit_width = T::get_required_bits();
-        // decompose e to bits
-        let e_id = self.define(e, statements_flattened);
-
-        // define variables for the bits
-        let e_bits_be: Vec<FlatVariable> = (0..bit_width).map(|_| self.use_sym()).collect();
-
-        // add a directive to get the bits
-        statements_flattened.push_back(FlatStatement::Directive(FlatDirective::new(
-            e_bits_be.clone(),
-            Solver::bits(bit_width),
-            vec![e_id],
-        )));
-
-        // bitness checks
-        for bit in e_bits_be.iter().take(bit_width) {
-            statements_flattened.push_back(FlatStatement::Condition(
-                FlatExpression::Identifier(*bit),
-                FlatExpression::Mult(
-                    box FlatExpression::Identifier(*bit),
-                    box FlatExpression::Identifier(*bit),
-                ),
-                RuntimeError::ConstantLtBitness,
-            ));
-        }
-
-        // bit decomposition check
-        let mut e_sum = FlatExpression::Number(T::from(0));
+        let e: FlatExpression<T> = self.define(e, statements_flattened).into();
```

Comment:
is this needed?

Reply:
I didn't think much of it because it was like that before, but I assume it's not needed here